src/libcore/future/mod.rs

```diff
@@ -2,6 +2,84 @@
 
 //! Asynchronous values.
 
+#[cfg(not(bootstrap))]
+use crate::{
+    ops::{Generator, GeneratorState},
+    pin::Pin,
+    ptr::NonNull,
+    task::{Context, Poll},
+};
+
 mod future;
 #[stable(feature = "futures_api", since = "1.36.0")]
 pub use self::future::Future;
+
+/// This type is needed because:
+///
+/// a) Generators cannot implement `for<'a, 'b> Generator<&'a mut Context<'b>>`, so we need to pass
```

Comment:
Why not? `impl<'a, 'b> Generator<&'a mut Context<'b>> for ...`

Reply:
Generator literals don't support that at the moment, see https://github.com/rust-lang/rust/issues/68923

Reply:
Can you leave a pointer to #68923 here?